### PR TITLE
Add YandexGPT embeddings models

### PR DESCRIPTION
**Pull request by tyumentsev4**
Introducing an ability to work with the YandexGPT chat model.

### CODE REVIEW

File: libs/langchain-community/src/embeddings/yandex.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  /**
+   * Yandex Cloud IAM token for service or user account
+   * with the `ai.languageModels.user` role.
+   */
+  iamToken?: string;
+}
+
+/**
+ * Class for generating embeddings using the YandexGPT Foundation models API. Extends the
+ * Embeddings class and implements GradientEmbeddingsParams and
```
Comment (bracesproul) [inline libs/langchain-community/src/embeddings/yandex.ts:37]:
```suggestion
 * Embeddings class and implements YandexGPTEmbeddingsParams and
```
nit

File: libs/langchain-community/src/embeddings/yandex.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    }
+  }
+
+  /**
+   * Method to generate embeddings for an array of documents.
+   * @param texts Array of documents to generate embeddings for.
+   * @returns Promise that resolves to a 2D array of embeddings for each document.
+   */
+  async embedDocuments(texts: string[]): Promise<number[][]> {
+    return this.caller.call(async () => {
```
Comment (bracesproul) [inline libs/langchain-community/src/embeddings/yandex.ts:94]: Small nit, in other embeddings classes we wrap the calls with an `embeddingWithRetry` method. Not exactly necessary in this case, but could prove helpful in the future.
[Example](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-mistralai/src/embeddings.ts#L133C3-L133C36)

**Review by bracesproul — COMMENTED** [review]
A few nits, also it would be helpful to have an integration test here (`embeddings/tests/yandex.int.test.ts`)

**Comment by tyumentsev4** [comment]
Check #3791

**Pull request closed without merging**